File: Day314_2025/Q314_Leetcode.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        // Q. Minimum Operations to Convert All Elements to Zero
        vector<int> s;
        int res = 0;
        for (int a : nums) {
            while (!s.empty() && s.back() > a) {
                s.pop_back();
            }
            if (a == 0) continue;
            if (s.empty() || s.back() < a) {
                ++res;
                s.push_back(a);
            }
        }
        return res;
    }
};

File: Day314_2025/Q314_Leetcode.py
from typing import List


class Solution:
    def minOperations(self, nums: List[int]) -> int:
        stack = []
        res = 0

        for a in nums:
            while stack and stack[-1] > a:
                stack.pop()
            if a == 0:
                continue
            if not stack or stack[-1] < a:
                res += 1
                stack.append(a)

        return res
